File: regions/system/library/go/building-blocks/secretstore.go
package buildingblocks

import "context"

// Secret represents a retrieved secret.
type Secret struct {
	Key      string            `json:"key"`
	Value    string            `json:"value"`
	Metadata map[string]string `json:"metadata,omitempty"`
}

// SecretStore provides unified secret retrieval.
type SecretStore interface {
	Component
	Get(ctx context.Context, key string) (*Secret, error)
	BulkGet(ctx context.Context, keys []string) ([]*Secret, error)
	Close(ctx context.Context) error
}

File: regions/system/library/typescript/building-blocks/src/secretstore.ts
import type { Component } from './component.js';

export interface SecretValue {
  key: string;
  value: string;
  metadata: Record<string, string>;
}

export interface SecretStore extends Component {
  getSecret(key: string): Promise<SecretValue>;
  bulkGet(keys: string[]): Promise<Record<string, SecretValue>>;
}
